src/entries/popup/hooks/useWatchPendingTransactions.ts

```diff
@@ -102,21 +102,30 @@ export const useWatchPendingTransactions = ({
         }
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
       } catch (e: any) {
+        const errorMessage = e?.message ?? String(e);
         logger.error(
           new RainbowError(
             `useWatchPendingTransaction: Failed to watch transaction`,
             { cause: e },
           ),
           {
-            message: e.message,
+            message: errorMessage,
             cause: e.cause,
             hash: tx.hash,
```

Comment:
e.cause and message will get picked up by the above logger cause; would also remove `hash` here to prevent deanonymizing the wallet